src/rule_tester.rs

```diff
@@ -40,7 +40,7 @@ impl RuleTester {
             "tmp.rs",
             ConfigBuilder::default()
                 .rule(&self.rule.meta().name)
-                .all_rules([self.rule.clone()])
+                .all_standalone_rules([self.rule.clone()])
                 .default_rule_configurations()
```

Comment:
This is getting weirder as an API now that it's "only applying" to the `all_standalone_rules` configuration (ie it's not creating any rule configurations for any plugin rules added via `.all_plugins()`)

I think it probably needs to decide if it's a "convenience API" basically just for the test-code consumers vs more general?